pyanaconda/install.py

```diff
@@ -51,74 +52,79 @@ def _writeKS(ksdata):
         f.write(str(ksdata))
 
 def doConfiguration(storage, payload, ksdata, instClass):
-    willWriteNetwork = not flags.flags.imageInstall and not flags.flags.dirInstall
-    willRunRealmd = ksdata.realm.discovered
-
-    # configure base, create users, configure addons, initramfs, post-install
-    step_count = 5
-
-    # network, maybe
-    if willWriteNetwork:
-        step_count += 1
-
-    # if a realm was discovered,
-    # increment the counter as the
-    # real joining step will be executed
-    if willRunRealmd:
-        step_count += 1
-
-    progress_init(step_count)
-
-    # Now run the execute methods of ksdata that require an installed system
-    # to be present first.
-    with progress_report(N_("Configuring installed system")):
-        ksdata.authconfig.execute(storage, ksdata, instClass)
-        ksdata.selinux.execute(storage, ksdata, instClass)
-        ksdata.firstboot.execute(storage, ksdata, instClass)
-        ksdata.services.execute(storage, ksdata, instClass)
-        ksdata.keyboard.execute(storage, ksdata, instClass)
-        ksdata.timezone.execute(storage, ksdata, instClass)
-        ksdata.lang.execute(storage, ksdata, instClass)
-        ksdata.firewall.execute(storage, ksdata, instClass)
-        ksdata.xconfig.execute(storage, ksdata, instClass)
-        ksdata.skipx.execute(storage, ksdata, instClass)
-
-    if willWriteNetwork:
-        with progress_report(N_("Writing network configuration")):
-            ksdata.network.execute(storage, ksdata, instClass)
-
-    # Creating users and groups requires some pre-configuration.
-    with progress_report(N_("Creating users")):
-        u = Users()
-        ksdata.rootpw.execute(storage, ksdata, instClass, u)
-        ksdata.group.execute(storage, ksdata, instClass, u)
-        ksdata.user.execute(storage, ksdata, instClass, u)
-        ksdata.sshkey.execute(storage, ksdata, instClass, u)
-
-    with progress_report(N_("Configuring addons")):
-        ksdata.addons.execute(storage, ksdata, instClass, u, payload)
-
-    with progress_report(N_("Generating initramfs")):
-        payload.recreateInitrds()
+    """Configure the installed system."""
+
+    configuration_queue = TaskQueue("Configuration queue")
+    # connect progress reporting
+    configuration_queue.queue_started.connect(lambda x: progress_message(x.status_message))
+    configuration_queue.queue_completed.connect(lambda x: progress_step("%s -- DONE" % x.status_message))
+
+    # schedule the execute methods of ksdata that require an installed system to be present
+    os_config = TaskQueue("Installed system configuration", N_("Configuring installed system"))
+    os_config.append(Task("Configure authconfig", lambda: ksdata.authconfig.execute(storage, ksdata, instClass)))
```

Comment:
I wouldn't like to see mixing of args and lambdas just because I have there kwargs. Is there any reason why not to create ``args=(), kwargs={}`` like for example in ``Thread.__init__()``.

https://docs.python.org/3/library/threading.html#threading.Thread